http-clients/url-connection-client/src/main/java/software/amazon/awssdk/http/urlconnection/UrlConnectionHttpClient.java

```diff
@@ -228,6 +239,85 @@ public HttpExecuteResponse call() throws IOException {
                                       .build();
         }
 
+        private Optional<OutputStream> tryGetOutputStream() {
+            return getAndHandle100Bug(() -> invokeSafely(connection::getOutputStream), Expect100BugHandlingBehavior.IGNORE);
+        }
+
+        private Optional<InputStream> tryGetInputStream() {
+            return getAndHandle100Bug(() -> invokeSafely(connection::getInputStream), Expect100BugHandlingBehavior.EXCEPTION);
+        }
+
+        private Optional<InputStream> tryGetErrorStream() {
+            return getAndHandle100Bug(() -> invokeSafely(connection::getErrorStream), Expect100BugHandlingBehavior.LOG);
+        }
+
+        private <T> Optional<T> getAndHandle100Bug(Supplier<T> supplier, Expect100BugHandlingBehavior bugBehavior) {
```

Comment:
Nice helper method! A small description of the bug may be helpful for posterity's sake.